flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointStorageLoader.java

```diff
@@ -44,6 +44,9 @@
 
     private static final String FILE_SYSTEM_STORAGE_NAME = "filesystem";
 
+    private static final String LEGACY_PRECEDENCE_LOG_MESSAGE =
+            "Legacy state backends can also be used as checkpoint storage and are configured to take precedence for backward-compatibility reasons.";
```

Comment:
```suggestion
            "Legacy state backends can also be used as checkpoint storage and take precedence for backward-compatibility reasons.";
```
Saying that they are "configured to do X" may imply that this behavior is configurable.